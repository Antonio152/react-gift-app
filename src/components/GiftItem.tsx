import React from 'react'
import { GiftContent } from '../types'

const GiftItem = ({ title, url }:GiftContent) => {
  return (
    <div className='card'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

export default GiftItem
