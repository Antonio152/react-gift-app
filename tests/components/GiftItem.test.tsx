import {render} from '@testing-library/react';
import React from 'react';
import GiftItem from '../../src/components/GiftItem';
describe('Testing de componente GiftItem', () => { 
    const title = 'NarutoImagen'
    const url = 'https://www.google.com/'
    const id = '1'

    test('Debe de hacer match con el Snapshot', () => { 
        const {container} = render(<GiftItem title={title} url={url} id={id}/>)
        expect(container).toMatchSnapshot()
    })
    test('Deberia Mostrar la URL de la imagen con el URL y el ALT indicado', () => { 
        const {getByRole} = render(<GiftItem title={title} url={url} id={id}/>)
        expect(getByRole('img')).toHaveProperty('src', url)
        expect(getByRole('img')).toHaveProperty('alt', title)
    })
    test('Deberia Mostrar el titulo de la imagen', () => { 
        const {getByText} = render(<GiftItem title={title} url={url} id={id}/>)
        expect(getByText(title)).toBeTruthy()
    })
    
    
 })