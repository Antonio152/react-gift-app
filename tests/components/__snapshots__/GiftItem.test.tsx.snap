// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing de componente GiftItem Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="NarutoImagen"
      src="https://www.google.com/"
    />
    <p>
      NarutoImagen
    </p>
  </div>
</div>
`;
